frontend/app/imageSelection/imageSelection.html

```diff
@@ -22,10 +22,10 @@
 
 <div id="newImageModal" class="modal">
     <div class="modal-content">
-        <h4>Upload Image</h4>
+        <h4>Upload New Image</h4>
         <p>Choose an image you want to upload</p>
-        <input type="file" id="newImage" name="newImage"/>
-        <a class="btn" ng-click="uploadImage()">Upload Image</a>
+        <input type="file" id="newImage" name="newImage" onchange="angular.element(this).scope().fileNameChanged()"/>
```

Comment:
Angular has it's own ng-change directive which could change this into `ng-change:"fileNameChanged()"`